dcm4che-net-audit/src/test/java/org/dcm4che3/net/audit/SyslogProtocolHandlerTest.java

```diff
@@ -0,0 +1,80 @@
+/* ***** BEGIN LICENSE BLOCK *****
+ * Version: MPL 1.1/GPL 2.0/LGPL 2.1
+ *
+ * The contents of this file are subject to the Mozilla Public License Version
+ * 1.1 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * http://www.mozilla.org/MPL/
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+ * for the specific language governing rights and limitations under the
+ * License.
+ *
+ * The Original Code is part of dcm4che, an implementation of DICOM(TM) in
+ * Java(TM), hosted at https://github.com/dcm4che.
+ *
+ * The Initial Developer of the Original Code is
+ * Agfa Healthcare.
+ * Portions created by the Initial Developer are Copyright (C) 2020
+ * the Initial Developer. All Rights Reserved.
+ *
+ * Contributor(s):
+ * See @authors listed below
+ *
+ * Alternatively, the contents of this file may be used under the terms of
+ * either the GNU General Public License Version 2 or later (the "GPL"), or
+ * the GNU Lesser General Public License Version 2.1 or later (the "LGPL"),
+ * in which case the provisions of the GPL or the LGPL are applicable instead
+ * of those above. If you wish to allow use of your version of this file only
+ * under the terms of either the GPL or the LGPL, and not to allow others to
+ * use your version of this file under the terms of the MPL, indicate your
+ * decision by deleting the provisions above and replace them with the notice
+ * and other provisions required by the GPL or the LGPL. If you do not delete
+ * the provisions above, a recipient may use your version of this file under
+ * the terms of any one of the MPL, the GPL or the LGPL.
+ *
+ * ***** END LICENSE BLOCK ***** */
+
+package org.dcm4che3.net.audit;
+
+import java.util.concurrent.ThreadPoolExecutor;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+
+import static org.hamcrest.core.Is.is;
+import static org.hamcrest.core.StringStartsWith.startsWith;
+import static org.junit.Assert.assertThat;
+
+/**
+ * @author Zhirong Liang <zhirong.liang@agfa.com>
+ */
+public class SyslogProtocolHandlerTest {
+
+    private static final String AUDIT_SYSLOG_THREAD_NAME_PREFIX = "Audit-Syslog-";
+    private static final int MINIMUM_NUMBER_OF_THREADS = 4;
+
+    @Test
+    public void executor_ThreadNamePrefixesWithCustomizedName_WhenNewThreadIsCreated() {
+
+        ThreadPoolExecutor executor = Whitebox.getInternalState(SyslogProtocolHandler.INSTANCE, "executor");

```

Comment:
Wouldn't it be nicer to use Mockito Spy or even making member protected and making private testable class that exposes the executor with public method? Typically unit testing would somehow test it via public methods. Is there another way to test it; maybe via calling execute?

Reply:
Good question. I don't think it's good idea to change the originl class. Maybe we don't have to unit test  ThreadPoolExecutor. But in order to address the issues that this artifact concerns. I added these 2 tests as I commented in another reply:

Todo :
1) use a named thread pool for auditing

2) verify the number of threads we configure